### PR TITLE
Add non-Euclidean Poisson system

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Add support for solving the Poisson equation on a background metric.

The reason I'm pushing this now is mostly for the second commit, which will be useful for @tomwlodarczyk's upcoming elasticity system.

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by nilsvu** [comment]
@fmahebert @prayush could you do the review for this? As always, feel free to forward to another reviewer if you don't have the time.

**Review by fmahebert — DISMISSED** [review]

**Comment by nilsvu** [comment]
@prayush please take a look.

**Review by prayush — DISMISSED** [review]

**Comment by nilsvu** [comment]
@sxs-collaboration/spectre-core-devs This PR is ready for a final look.
Thanks for reviewing @fmahebert @prayush. :)

File: src/Elliptic/Systems/Poisson/Equations.hpp
Comment (kidder) [inline src/Elliptic/Systems/Poisson/Equations.hpp]: `non_euclidean_fluxes` to be consistent with `NonEuclideanFluxes`

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/Poisson/Test_Lorentzian.cpp
Comment (kidder) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/Poisson/Test_Lorentzian.cpp]: please make the test run faster

**Review by kidder — CHANGES_REQUESTED** [review]

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/Poisson/Test_Lorentzian.cpp
Comment (nilsvu) [reply tests/Unit/PointwiseFunctions/AnalyticSolutions/Poisson/Test_Lorentzian.cpp]: this was left over from an early issue, deleted the line and squashed the change. thanks for catching this

**Comment by nilsvu** [comment]
done @kidder. Thanks for reviewing!

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/Poisson/Test_Lorentzian.cpp
Comment (kidder) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/Poisson/Test_Lorentzian.cpp]: this should be in the previous commit...

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash this

**Comment by nilsvu** [comment]
done @kidder

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@prayush @fmahebert there were some fixups

**Review by fmahebert — APPROVED** [review]

**Comment by nilsvu** [comment]
@prayush please take a quick look

**Comment by nilsvu** [comment]
@kidder do you want to wait for @prayush to re-approve this (he approved earlier)? If so, @prayush could you please take a look?

**Comment by kidder** [comment]
@prayush this just needs your final approval

**Comment by nilsvu** [comment]
@prayush please take a look...

**Comment by kidder** [comment]
@prayush please file an issue if you were not happy with this

**Pull request merged**
